src/corehost/corehost.cpp

```diff
@@ -293,9 +293,20 @@ int main(const int argc, const pal::char_t* argv[])
 #if defined(_WIN32) && defined(FEATURE_APPHOST)
     if (get_windows_graphical_user_interface_bit())
     {
-        // If this is a GUI application, buffer errors to display them later. Without this any errors are effectively lost
-        // unless the caller explicitly redirects stderr. This leads to bad experience of running the GUI app and nothing happening.
-        trace::set_error_writer(buffering_trace_writer);
+        pal::string_t env_value;
+        bool gui_errors_disabled = false;
+
+        if (pal::getenv(_X("DOTNET_DISABLE_GUI_ERRORS"), &env_value))
+        {
+            gui_errors_disabled = pal::xtoi(env_value.c_str()) == 1;
```

Comment:
This should probably be `!= 0` rather than `== 1`, but the value probably doesn't matter at all....

Could be 
```C++
bool gui_errors_disabled = pal::getenv(_X("DOTNET_DISABLE_GUI_ERRORS"), &env_value);
```

Reply:
This was modeled after `DOTNET_MULTILEVEL_LOOKUP` which also compares to exactly `1` only. I think it's safer that way:
* `0` - should mean disable, so that it's possible to have the variable set always (in some environments it's pretty tricky to unset a variable)
* `2` and more - I guess we could allow these, but I would rather be strict for now, so that in the future we can use these for something else if necessary.